owncloudApp/src/main/java/com/owncloud/android/presentation/transfers/TransferListComposeFragment.kt

```diff
@@ -0,0 +1,532 @@
+                        }
+                    }
+            }
+        }
+    }
+
+    @Composable
+    fun TransferItem(transfer: OCTransfer, space: OCSpace?, workInfo: List<WorkInfo>) {
+        val remoteFile = File(transfer.remotePath)
+        var fileName = remoteFile.name
+        if (fileName.isEmpty()) {
+            fileName = File.separator
+        }
+
+        var path = "path/to/file"
+        remoteFile.parent?.let {
+            path = if (it.endsWith("${OCFile.PATH_SEPARATOR}")) {
+                it
+            } else {
+                "$it${OCFile.PATH_SEPARATOR}"
+            }
+        }
+
+        Row(
+            modifier = Modifier
+                .layoutId("LisItemLayout")
+                .fillMaxWidth()
+                .padding(top = 8.dp, bottom = 8.dp)
+                .clickable(enabled = transfer.status == TransferStatus.TRANSFER_FAILED) { retryUpload(transfer) },
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Box(
+                modifier = Modifier
+                    .width(60.dp)
+                    .height(72.dp),
+                contentAlignment = Alignment.Center
+            ) {
+                Image(
+                    modifier = Modifier
+                        .layoutId("Thumbnail")
+                        .width(32.dp)
+                        .height(32.dp),
+                    painter = painterResource(
+                        id = MimetypeIconUtil.getFileTypeIconId(
+                            MimetypeIconUtil.getBestMimeTypeByFilename(transfer.localPath),
+                            fileName
+                        )
+                    ),
+                    contentDescription = "Thumbnail"
+                )
+            }
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .weight(1f)
+            ) {
+                Text(
+                    modifier = Modifier
+                        .layoutId("upload_name"),
+                    overflow = TextOverflow.Ellipsis,
+                    maxLines = 1,
+                    color = colorResource(id = R.color.textColor),
+                    fontSize = 16.sp,
+                    text = fileName
+                )
+                Row {
+                    Text(
+                        modifier = Modifier
+                            .layoutId("upload_file_size"),
+                        color = colorResource(id = R.color.list_item_lastmod_and_filesize_text),
+                        overflow = TextOverflow.Ellipsis,
+                        maxLines = 1,
+                        fontSize = 12.sp,
+                        text = DisplayUtils.bytesToHumanReadable(transfer.fileSize, requireContext())
+                    )
+                    if (transfer.transferEndTimestamp != null && transfer.status != TransferStatus.TRANSFER_FAILED) {
+                        transfer.transferEndTimestamp?.let {
```

Comment:
You already checked if this is null. If we're at this point is because `transferEndTimestamp` is not null

Reply:
Changed 👍